packages/agreement-process/src/services/agreementService.ts

```diff
@@ -11,4 +63,49 @@ export const agreementService = {
     const agreement = await readModelService.readAgreementById(agreementId);
     return agreement?.data;
   },
+  async deleteAgreementById(
+    agreementId: string,
+    authData: AuthData
+  ): Promise<void> {
+    const agreement = await readModelService.readAgreementById(agreementId);
+
+    await repository.createEvent(
+      await deleteAgreementLogic({
+        agreementId,
+        authData,
+        deleteFile: fileManager.deleteFile,
+        agreement,
+      })
+    );
+  },
 };
+
+export async function deleteAgreementLogic({
+  agreementId,
+  authData,
+  deleteFile,
+  agreement,
+}: {
+  agreementId: string;
+  authData: AuthData;
+  deleteFile: (path: string) => Promise<void>;
+  agreement: WithMetadata<PersistentAgreement> | undefined;
+}): Promise<CreateEvent<AgreementEvent>> {
+  assertAgreementExist(agreementId, agreement);
+  assertRequesterIsConsumer(agreement.data.consumerId, authData.organizationId);
+
+  const deletableStates: PersistentAgreementState[] = [
+    persistentAgreementState.draft,
+    persistentAgreementState.missingCertifiedAttributes,
+  ];
+
+  if (!deletableStates.includes(agreement.data.state)) {
+    throw agreementNotInExpectedState(agreementId, agreement.data.state);
+  }
+
+  for (const d of agreement.data.consumerDocuments) {
+    await deleteFile(d.path);
+  }
```

Comment:
ℹ️ I don't know how Delete method works in scala repository 
but in this case could be useful to use Promise.any to log rejected promises?
```suggestion
const documentToDelete = agreement.data.consumerDocuments.map(d => deleteFile(d.path))
const results = await Promise.any(documentToDelete);

const rejectedPromise = results.filter(p => p.status = "rejected")
logger.warn(rejectedPromise)
```

Reply:
Maybe we could let the `fileManager` log the failed requests. In this way, we have a centralized way to control the log for file operations. In this case, I will delegate the logging operation to a dedicated PR that improves the `FileManager` implementation.
Let me know what you think.